pkg/agent/route/route_linux.go

```diff
@@ -278,6 +287,10 @@ func (c *Client) Initialize(nodeConfig *config.NodeConfig, done func()) error {
 	if c.networkConfig.TrafficEncryptionMode == config.TrafficEncryptionModeWireGuard {
 		c.initWireguard()
 	}
+	if c.nodeLatencyMonitorEnabled {
```

Comment:
Should it only add the rule when `latencyConfig.Enable` in pkg/agent/monitortool/monitor.go is `true`?
If we only check the feature gate, the 4 rules would be installed unconditionally on all clusters even the API is not used, which might add unnecessary overhead to Node-to-Node traffic. This might already be the case of nodeNetworkPolicyEnabled, but with more feature-specific rule added, I'm a bit concerned about the total overhead.

Or we could defer this and add one item to these features' beta graduation criteria: the feature-specific rules should only be installed when the feature is in-use.

what do you think? @antoninbas

Reply:
I am fine with taking care of this as a follow-up item, and prior to graduating the feature. Maybe @Dhruv-J could open a separate issue to keep track of this.